### PR TITLE
Add test_amd_mi250 CI job including ROCm matmul tests.

### PR DESCRIPTION
**Pull request by erman-gurses**
Progress on https://github.com/iree-org/iree/issues/17159

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -389,6 +389,45 @@ jobs:
...
+      - name: "Extracting install directory"
+        run: tar -xf "${INSTALL_DIR_ARCHIVE}"
+      - name: "Building tests"
+        run: |
+          ./build_tools/pkgci/build_tests_using_package.sh ${INSTALL_DIR}
+      - name: "Running GPU tests"
+        env:
+          IREE_CTEST_LABEL_REGEX: ^requires-gpu|^driver=hip$
+          IREE_NVIDIA_SM80_TESTS_DISABLE: 1
+          IREE_MULTI_DEVICE_TESTS_DISABLE: 1
```
Comment (ScottTodd) [inline .github/workflows/ci.yml]: We should be able to flip this to `0` (enable multi-device tests). Let's leave that for a follow-up - maybe @sogartar can try it.

File: runtime/src/iree/hal/drivers/hip/cts/CMakeLists.txt [hunk 2]
```diff
@@ -59,6 +60,9 @@ iree_hal_cts_test_suite(
   DEPS
     iree::hal::drivers::hip::registration
   EXCLUDED_TESTS
+    "command_buffer"
+    "command_buffer_dispatch"
+    "file"
```
Comment (ScottTodd) [inline runtime/src/iree/hal/drivers/hip/cts/CMakeLists.txt]: Add comment on why these tests are excluded:

```suggestion
    # These tests fail with:
    #     UNAVAILABLE; missing hipDrvGraphAddMemcpyNode symbol;
    #     cannot use graph-based command buffer
    "command_buffer"
    "command_buffer_dispatch"
    "file"
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -389,6 +389,45 @@ jobs:
                 "./build_tools/scripts/check_cuda.sh
                 ./build_tools/scripts/check_vulkan.sh
                 ./build_tools/cmake/ctest_all.sh ${BUILD_DIR}"
+
```
Comment (ScottTodd) [inline .github/workflows/ci.yml]: remove trailing whitespace here (you can configure your editor to do this for you)

Comment (ScottTodd) [reply .github/workflows/ci.yml]: (see the failing lint check: https://github.com/iree-org/iree/actions/runs/8976316866/job/24652921251?pr=17293)

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -389,6 +389,45 @@ jobs:
...
+          IREE_CTEST_LABEL_REGEX: ^requires-gpu|^driver=hip$
+          IREE_NVIDIA_SM80_TESTS_DISABLE: 1
+          IREE_MULTI_DEVICE_TESTS_DISABLE: 1
+          IREE_AMD_RDNA3_TESTS_DISABLE: 1
+          IREE_NVIDIA_GPU_TESTS_DISABLE: 0
+          IREE_CUDA_DISABLE: 1
+          IREE_CPU_DISABLE: 1
+          IREE_HIP_DISABLE: 0
+        run: |
+          ./build_tools/cmake/ctest_all.sh ${BUILD_DIR}
```
Comment (ScottTodd) [inline .github/workflows/ci.yml:430]: Test output looks good: https://github.com/iree-org/iree/actions/runs/8976316871/job/24653225899#step:7:22

```
100% tests passed, 0 tests failed out of 124

Label Time Summary:
driver=hip                      = 650.21 sec*proc (124 tests)
iree/hal/drivers/hip            =   3.08 sec*proc (1 test)
iree/hal/drivers/hip/cts        =  98.17 sec*proc (21 tests)
iree/tests/e2e/matmul           =  13.86 sec*proc (2 tests)
iree/tests/e2e/stablehlo_ops    = 340.22 sec*proc (61 tests)
iree/tests/e2e/tosa_ops         = 194.88 sec*proc (39 tests)
noasan                          = 548.96 sec*proc (102 tests)
nomsan                          = 548.96 sec*proc (102 tests)
notsan                          = 548.96 sec*proc (102 tests)
noubsan                         = 548.96 sec*proc (102 tests)
requires-gpu-amd                = 583.60 sec*proc (112 tests)
requires-gpu-gfx908+            =  13.86 sec*proc (2 tests)
```

File: tests/e2e/matmul/CMakeLists.txt [hunk 5]
```diff
@@ -2098,3 +2041,68 @@ iree_generated_e2e_runner_test(
...
+    "--shapes=gpu_large_aligned"
+    "--compilation_info=LLVMGPUVectorDistributeMFMA"
+  TEST_RUNNER
+    iree_tools_testing_e2e_iree-e2e-matmul-test
+  TARGET_BACKENDS
+    "rocm"
+  DRIVERS
+    "hip"
+  COMPILER_FLAGS
+    "--iree-rocm-target-chip=gfx942"
```
Comment (ScottTodd) [inline tests/e2e/matmul/CMakeLists.txt]: Use `IREE_HIP_TEST_COMPILER_FLAGS` here

File: tests/e2e/matmul/CMakeLists.txt [hunk 6]
```diff
@@ -2098,3 +2041,68 @@ iree_generated_e2e_runner_test(
...
```
Comment (ScottTodd) [inline tests/e2e/matmul/CMakeLists.txt]: and here

File: tests/e2e/matmul/CMakeLists.txt [hunk 7]
```diff
@@ -2098,3 +2041,68 @@ iree_generated_e2e_runner_test(
...
+if(IREE_HIP_TEST_TARGET_CHIP MATCHES "^gfx")
+
+unset(IREE_HIP_TEST_COMPILER_FLAGS)
+list(APPEND IREE_HIP_TEST_COMPILER_FLAGS
+  "--iree-rocm-target-chip=${IREE_HIP_TEST_TARGET_CHIP}"
+)
+
+iree_generated_e2e_runner_test(
+  NAME
+    e2e_matmul_rocm_f16_large_cdna3_mfma
```
Comment (ScottTodd) [inline tests/e2e/matmul/CMakeLists.txt:2054]: Looks like these are failing now
```
117/124 Test  #163: iree/tests/e2e/matmul/e2e_matmul_rocm_f16_large_cdna3_mfma_rocm_hip ........***Failed    5.54 sec
iree/runtime/src/iree/hal/drivers/hip/native_executable.c:184: FAILED_PRECONDITION; HIP driver error 'hipErrorSharedObjectInitFailed' (303): shared object initialization failed; mismatched target chip? missing/wrong bitcode directory?; while invoking native function hal.executable.create; while calling import; 

The following tests FAILED:
	163 - iree/tests/e2e/matmul/e2e_matmul_rocm_f16_large_cdna3_mfma_rocm_hip (Failed)
	164 - iree/tests/e2e/matmul/e2e_matmul_rocm_f16_large_cdna3_mfma_tb_rocm_hip (Failed)
```

Comment (erman-gurses) [reply tests/e2e/matmul/CMakeLists.txt:2054]: Thanks, I updated the compiler flag. Looks like they are passing now.

**Review by ScottTodd — APPROVED** [review]

File: runtime/src/iree/hal/drivers/hip/cts/CMakeLists.txt
Comment (ScottTodd) [inline runtime/src/iree/hal/drivers/hip/cts/CMakeLists.txt:1]: semaphore submission test segfaulted on postsubmit 🤔 
https://github.com/iree-org/iree/actions/runs/8979325892/job/24661385410#step:7:161
```
 12/124 Test   #75: iree/hal/drivers/hip/cts/hip_graph_semaphore_submission_test ...............***Exception: SegFault  2.95 sec
[==========] Running 14 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 14 tests from CTS/semaphore_submission_test
[ RUN      ] CTS/semaphore_submission_test.SubmitWithNoCommandBuffers/hip
[       OK ] CTS/semaphore_submission_test.SubmitWithNoCommandBuffers/hip (2773 ms)
[ RUN      ] CTS/semaphore_submission_test.SubmitAndSignal/hip
[       OK ] CTS/semaphore_submission_test.SubmitAndSignal/hip (1 ms)
[ RUN      ] CTS/semaphore_submission_test.SubmitWithWait/hip
[       OK ] CTS/semaphore_submission_test.SubmitWithWait/hip (1 ms)
[ RUN      ] CTS/semaphore_submission_test.SubmitWithMultipleSemaphores/hip
[       OK ] CTS/semaphore_submission_test.SubmitWithMultipleSemaphores/hip (1 ms)
[ RUN      ] CTS/semaphore_submission_test.WaitAllHostAndDeviceSemaphores/hip
[       OK ] CTS/semaphore_submission_test.WaitAllHostAndDeviceSemaphores/hip (1 ms)
[ RUN      ] CTS/semaphore_submission_test.WaitAnyHostAndDeviceSemaphoresAndDeviceSignals/hip
[       OK ] CTS/semaphore_submission_test.WaitAnyHostAndDeviceSemaphoresAndDeviceSignals/hip (1 ms)
[ RUN      ] CTS/semaphore_submission_test.WaitAnyHostAndDeviceSemaphoresAndHostSignals/hip
[       OK ] CTS/semaphore_submission_test.WaitAnyHostAndDeviceSemaphoresAndHostSignals/hip (1 ms)
[ RUN      ] CTS/semaphore_submission_test.IntermediateSemaphoreBetweenDeviceBatches/hip
[       OK ] CTS/semaphore_submission_test.IntermediateSemaphoreBetweenDeviceBatches/hip (1 ms)
[ RUN      ] CTS/semaphore_submission_test.TwoBatchesWaitingOn1FormerBatchAmongst2/hip
[       OK ] CTS/semaphore_submission_test.TwoBatchesWaitingOn1FormerBatchAmongst2/hip (1 ms)
[ RUN      ] CTS/semaphore_submission_test.TwoBatchesWaitingOnDifferentSemaphoreValues/hip
[       OK ] CTS/semaphore_submission_test.TwoBatchesWaitingOnDifferentSemaphoreValues/hip (1 ms)
[ RUN      ] CTS/semaphore_submission_test.BatchWaitingOnAnotherAndHostSignal/hip
[       OK ] CTS/semaphore_submission_test.BatchWaitingOnAnotherAndHostSignal/hip (0 ms)
[ RUN      ] CTS/semaphore_submission_test.DeviceBatchSignalAnotherAndHost/hip

 13/124 Test  #688: iree/tests/e2e/stablehlo_ops/check_hip_stream_dot.mlir .....................   Passed    2.91 sec
```

**Pull request merged**